x/datapool/keeper/pool.go

```diff
@@ -129,7 +129,7 @@ func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams t
 	// pool address for deposit
 	poolAddress, err := sdk.AccAddressFromBech32(newPoolAddr)
 	if err != nil {
-		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address of pool %s", newPoolAddr)
+		return 0, sdkerrors.Wrapf(err, "invalid address of pool %s", newPoolAddr)
```

Comment:
We need to define how to respond to the error.

Some places respond simply by appending a string.
Another place defines and uses the error type in the location below.
https://github.com/medibloc/panacea-core/blob/master/x/datapool/types/errors.go

I usually define the error type and append the error details after it.
Is there any good way?

Reply:
If there is an error from functions (like `err` from `sdk.AccAddressFromBech32()`) and we check the `err` is not nil, I usually return the `err` wrapping with additional information to inform where the `err` comes from.

Otherwise we use custom logic for checking error case ([example](https://github.com/medibloc/panacea-core/blob/ba80e32aab61b1c9240bd08e8ff8e61b2d91913b/x/datapool/keeper/pool.go#L159)), I defined sentinel error and return it wrapping with some additional information.

I referred this post, https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully. Maybe @youngjoon-lee can give some advices about this.

Reply:
It depends on whether or not you want to expose the specified sentinel error to the function caller with ignoring whatever the child function returns.

If you think no caller cares about sentinel errors, you can just return a wrapped error. In this case, you actually don't need to use `sdkerrors.Wrapf()`. You can just use the `fmt.Errorf("%v", ...)`.

But, in general, we've returned wrapped sentinel errors in `x/.../keeper/...` packages, even though not all callers expect sentinel errors. So, I think it's better to follow that style for now. Honestly, I don't like the `sdkerrors` stuff.

Reply:
Ok! I'll change it as gyuguen did. 🙏